hr_timesheet_purchase_order/models/res_partner.py

```diff
@@ -0,0 +1,252 @@
+        [
+            ("day", "Days"),
+            ("week", "Weeks"),
+            ("month", "Months"),
+            ("year", "Years"),
+        ],
+        default="week",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_type = fields.Selection(
+        [
+            ("forever", "Forever"),
+            ("until", "End Date"),
+            ("after", "Number of Repetitions"),
+        ],
+        default="forever",
+        string="Until",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_until = fields.Date(
+        string="End Date", compute="_compute_repeat", readonly=False
+    )
+    repeat_number = fields.Integer(
+        string="Repetitions", default=1, compute="_compute_repeat", readonly=False
+    )
+
+    repeat_on_month = fields.Selection(
+        [
+            ("date", "Date of the Month"),
+            ("day", "Day of the Month"),
+        ],
+        default="date",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    repeat_on_year = fields.Selection(
+        [
+            ("date", "Date of the Year"),
+            ("day", "Day of the Year"),
+        ],
+        default="date",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    mon = fields.Boolean(string="Mon", compute="_compute_repeat", readonly=False)
+    tue = fields.Boolean(string="Tue", compute="_compute_repeat", readonly=False)
+    wed = fields.Boolean(string="Wed", compute="_compute_repeat", readonly=False)
+    thu = fields.Boolean(string="Thu", compute="_compute_repeat", readonly=False)
+    fri = fields.Boolean(string="Fri", compute="_compute_repeat", readonly=False)
+    sat = fields.Boolean(string="Sat", compute="_compute_repeat", readonly=False)
+    sun = fields.Boolean(string="Sun", compute="_compute_repeat", readonly=False)
+
+    repeat_day = fields.Integer(
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    @api.onchange("repeat_day", "repeat_month")
+    def _onchange_repeat_day(self):
+        if 0 > self.repeat_day or self.repeat_day > 31:
+            self.repeat_day = 1
+        if self.repeat_month == "february" and self.repeat_day > 29:
+            self.repeat_day = 28
+
+    repeat_week = fields.Selection(
+        [
+            ("first", "First"),
+            ("second", "Second"),
+            ("third", "Third"),
+            ("last", "Last"),
+        ],
+        default="first",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_weekday = fields.Selection(
+        [
+            ("mon", "Monday"),
+            ("tue", "Tuesday"),
+            ("wed", "Wednesday"),
+            ("thu", "Thursday"),
+            ("fri", "Friday"),
+            ("sat", "Saturday"),
+            ("sun", "Sunday"),
+        ],
+        string="Day Of The Week",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_month = fields.Selection(
+        [
```

Comment:
The same selection in prev model, would you create variable for this selection and use it in fields?

Reply:
Selection argument is static data. Just create variable `MONTH_SELECTION`
```py
MONTH_SELECTION = [
("january", "January"),
("february", "February"),
....
]

...

repeat_month = fields.Selection(selection=MONTH_SELECTION)
```
The same for week day field